src/main/java/everyos.browser.webicitybrowser/everyos/browser/webicitybrowser/gui/ui/OverlyingBlockComponentUI.java

```diff
@@ -1,72 +1,103 @@
 package everyos.browser.webicitybrowser.gui.ui;
 
+import everyos.browser.javadom.imp.TimeUtil;
 import everyos.browser.webicitybrowser.gui.Styling;
+import everyos.browser.webicitybrowser.gui.component.CircularText;
 import everyos.browser.webicitybrowser.gui.component.OverlyingBlockComponent;
+import everyos.browser.webicitybrowser.util.TimeSystem;
 import everyos.engine.ribbon.core.component.Component;
+import everyos.engine.ribbon.core.directive.PositionDirective;
+import everyos.engine.ribbon.core.directive.SizeDirective;
+import everyos.engine.ribbon.core.event.UIEvent;
 import everyos.engine.ribbon.core.rendering.Renderer;
+import everyos.engine.ribbon.core.shape.Dimension;
+import everyos.engine.ribbon.core.shape.Location;
 import everyos.engine.ribbon.core.shape.Rectangle;
+import everyos.engine.ribbon.core.shape.SizePosGroup;
 import everyos.engine.ribbon.core.ui.ComponentUI;
+import everyos.engine.ribbon.core.ui.UIDirective;
 import everyos.engine.ribbon.core.ui.UIManager;
 import everyos.engine.ribbon.ui.simple.SimpleBlockComponentUI;
+import everyos.engine.ribbon.ui.simple.appearence.Appearence;
+import everyos.engine.ribbon.ui.simple.helper.ComputedChildrenHelper;
+import everyos.engine.ribbon.ui.simple.helper.StringWrapHelper;
+import everyos.engine.ribbon.ui.simple.layout.InlineBlockLayout;
+import everyos.engine.ribbon.ui.simple.layout.InlineLayout;
 
 public class OverlyingBlockComponentUI extends SimpleBlockComponentUI {
 
+    private Appearence appearence;
     private OverlyingBlockComponent component;
-    private Component contentPane;
-    private ComponentUI contentPaneUI;
+    private InlineBlockLayout layout;
 
     public OverlyingBlockComponentUI(Component c, ComponentUI parent) {
         super(c, parent);
-    }
-
-    //TODO: Fix merge conflicts
-    
-    /*@Override
-    protected ComponentUI[] calcChildren(UIManager uimgr) {
-        component = getComponent();
-        if (component.getChildren().length < 1) return new ComponentUI[0];
-        contentPane = component.getChildren()[0];
-        contentPane.unbindAll();
-        contentPaneUI = uimgr.get(contentPane, getParent());
-        contentPane.bind(contentPaneUI);
-        return new ComponentUI[]{contentPaneUI};
+        component = (OverlyingBlockComponent) c;
```

Comment:
We also have a method to do this. .casted(Type.class)

Reply:
Having long chaining methods is not always a good idea. But yeah, casted does it's job good.

Actually... You can make getComponent itself a generic. So you won't even have to pass in the class object.

Reply:
I had that previously
https://github.com/WebicityBrowser/Webicity/blob/71d90ace7bb2c454e9f480cd3096ff33b1e6c92b/src/main/java/everyos.engine.ribbon.ui.simple/everyos/engine/ribbon/ui/simple/SimpleComponentUI.java#L83
I think it disappeared when I was refactoring

Reply:
Not fixing this issue, as a lot of classes have to be made Generic (all UI classes).
While that's not a bad thing, it's a problem for UIManager. Which only yields Component and not a possible Generic.
That can still be fixed by casting, but that's not a good idea, as we already have the casted method.